sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -566,7 +563,14 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
     // Filter out test files with invalid extensions such as temp files created
     // by vi (.swp), Mac (.DS_Store) etc.
     val filteredFiles = files.filter(_.getName.endsWith(validFileExtensions))
-    filteredFiles ++ dirs.flatMap(listFilesRecursively)
+    val allFiles = filteredFiles ++ dirs.flatMap(listFilesRecursively)
+    // SPARK-32106 Since we add SQL test 'transform.sql' will use `cat` command,
+    // here we need to check command available
+    if (TestUtils.testCommandAvailable("/bin/bash")) {
+      allFiles
+    } else {
+      allFiles.filterNot(_.getName == "transform.sql")
```

Comment:
Could you use `ignore` instead of filtering out it?

Reply:
Yeah, can we call `assume` to trigger skipping the test instead of filtering out here?

Reply:
`assume` makes `SQLQueryTestSuite` exit

Reply:
Can you double check and investigate to make it working @beliefer? `assume` under `test` should work and skip only that single test.

Reply:
Yes. the origin code will skip the whole `SQLQueryTestSuite`.

Reply:
Does that skip all tests only in `SQLQueryTestSuite`? or does it happen in other suites too? e.g.) `SparkScriptTransformationSuite` or `PipedRDDSuite`?

Reply:
This skipping behavior not happen in `SparkScriptTransformationSuite` and `PipedRDDSuite`. Because `SparkScriptTransformationSuite` and `PipedRDDSuite` write each test case in single `SQLTestUtils.test`.
But `SQLQueryTestSuite` calls `TestUtils.testCommandAvailable("/bin/bash")` in common place.

Reply:
Can we conditionally call `assume` based on the name of test?

Reply:
e.g.) `assume(... test name contains transform ... || TestUtils.testCommandAvailable("/bin/bash"))`

Reply:
`SQLQueryTestSuite` must contains `transform.sql`. Why we need to judge `test name contains transform`?

Reply:
I meant the fix such as:

```scala
    assume(
      !testCase.inputFile.endsWith("transform.sql") ||
      TestUtils.testCommandAvailable("/bin/bash"))
```

I tested that it only skips transform.sql when `/bin/bash` is not available:

```
[info] - transform.sql !!! CANCELED !!! (36 milliseconds)
[info]   "/.../spark/sql/core/src/test/resources/sql-tests/inputs/transform.sql" ended with "transform.sql", and org.apache.spark.TestUtils.testCommandAvailable("/bin/bas") was false (SQLQueryTestSuite.scala:265)
[info]   org.scalatest.exceptions.TestCanceledException:
[info]   at org.scalatest.Assertions.newTestCanceledException(Assertions.scala:475)
[info]   at org.scalatest.Assertions.newTestCanceledException$(Assertions.scala:474)
[info]   at org.scalatest.Assertions$.newTestCanceledException(Assertions.scala:1231)
[info]   at org.scalatest.Assertions$AssertionsHelper.macroAssume(Assertions.scala:1310)
[info]   at org.apache.spark.sql.SQLQueryTestSuite.runTest(SQLQueryTestSuite.scala:265)
[info]   at org.apache.spark.sql.SQLQueryTestSuite.$anonfun$createScalaTestCase$5(SQLQueryTestSuite.scala:247)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:190)
[info]   at org.apache.spark.SparkFunSuite.withFixture(SparkFunSuite.scala:176)
```

Reply:
If so, `SQLQueryTestSuite` will execute `TestUtils.testCommandAvailable("/bin/bash")` many times.

Reply:
`TestUtils.testCommandAvailable("/bin/bash")` won't be executed via short circuiting

Reply:
If the file is `transform.sql`, `TestUtils.testCommandAvailable("/bin/bash")` return false. Then `assume` will break the follow test cases.

Reply:
The case I tested above is when the file is `transform.sql` and  `TestUtils.testCommandAvailable("/bin/bash")` returns `false`. I tested it by manually changing `/bin/bash` to an arbitrary executable such as `/bin/bas`.

Does it fail and stop all other test cases? Then maybe it is a Windows specific behaviour. If that's the case, I am fine with the current fix.

Reply:
Another reason is each test case will execute `assume`, if we adopt `ignoreList` to fix this issue, it seems simpler.